src/manifest.rs

```diff
@@ -78,30 +83,53 @@ pub struct Manifest {
 
 impl Manifest {
     /// Checks if the manifest file exists in the filesystem
-    pub async fn exists() -> eyre::Result<bool> {
+    pub async fn exists() -> miette::Result<bool> {
         fs::try_exists(MANIFEST_FILE)
             .await
-            .wrap_err("Failed to detect manifest")
+            .into_diagnostic()
+            .wrap_err_with(|| FileExistsError(MANIFEST_FILE))
     }
 
     /// Loads the manifest from the current directory
-    pub async fn read() -> eyre::Result<Self> {
-        let toml = fs::read_to_string(MANIFEST_FILE)
-            .await
-            .wrap_err("Failed to read manifest")?;
-
-        let raw: RawManifest = toml.parse().wrap_err("Failed to parse manifest")?;
+    pub async fn read() -> miette::Result<Self> {
+        Self::read_from(MANIFEST_FILE).await
+    }
 
-        Ok(raw.into())
+    /// Loads the manifest from the given path
+    pub async fn read_from(path: impl AsRef<Path>) -> miette::Result<Self> {
```

Comment:
can we merge them again? this isnt used anywhere right now, correct?

Reply:
It is, actually, in `PackageStore::resolve()`.

Reply:
Ah! Oka, well, valid reasoning :)